src/v/serde/serde.h

```diff
@@ -403,6 +411,8 @@ void write(iobuf& out, T t) {
         }
         auto d = std::chrono::duration_cast<std::chrono::nanoseconds>(t);
         write<int64_t>(out, d.count());
+    } else if constexpr (is_time_point_v<Type>) {
+        write(out, t.time_since_epoch());
```

Comment:
This doesn't seem to cast the underlying type. Is this in `std::nanoseconds` or `duration::rep`?

Reply:
It doesn't matter, right? (unless I'm misunderstanding your question). Its like serializing any duration, the cast to ns happens in the is_duration_v<> branch.

Reply:
`t.time_since_epoch()` will serialise in units of `std::duration_cast<T::rep>`. For `std::chrono::duration` we chose `std::chrono::nanonseconds`. So I wonder why the disparity.

Reply:
Oh, I see, it's forwarded to the `std::duration` overload?

Reply:
Right. Eventually it is casted to std::chrono::nanoseconds.